packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,359 @@
+
+_Component Name_
+
+-   `nimble-table-column-mapping`
+
+_Props/Attrs_
+
+-   `field-name`: string
+-   `data-type`: 'string' | 'number' | 'boolean' | null (defaults to 'string')
+-   `fractional-width`: number (defaults to 1)
+-   `min-pixel-width`: number (defaults to minimum supported by table)
+
+_Content_
+
+-   column title (text or icon)
+-   1 or more `nimble-mapping-text` elements
+
+#### Mapping element (icon):
+
+_Component Name_
+
+-   `nimble-mapping-icon`
+
+_Props/Attrs_
+
+-   `key`: string | number | boolean | null
+-   `icon`: string - name of the Nimble icon element
+-   `severity`: string - one of the supported enum values. Controls color of the icon.
+-   `label`: string - localized value used as the accessible name and `title` of the icon. Will also be displayed in the group header.
+-   `default`: boolean - presence causes this mapping to be used when no others match the value
+
+#### Mapping element (spinner):
+
+_Component Name_
+
+-   `nimble-mapping-spinner`
+
+_Props/Attrs_
+
+-   `key`: string | number | boolean | null
+-   `label`: string - localized value used as the accessible name and `title` of the spinner. Will also be displayed in the group header.
+-   `default`: boolean - presence causes this mapping to be used when no others match the value
+
+#### Mapping element (text):
+
+_Component Name_
+
+-   `nimble-mapping-text`
+
+_Props/Attrs_
+
+-   `key`: string | number | boolean | null
+-   `label`: string - display text
+-   `default`: boolean - presence causes this mapping to be used when no others match the value
+
+### Anatomy
+
+#### `nimble-table-column-mapping`
+
+```HTML
+<template slot="${x => x.columnInternals.uniqueId}">
+    <slot
+        ${slotted('mappings')}
+        name="mapping">
+    </slot>
+    <span class="header-content">
+        <slot></slot>
+    </span>
+</template>
+```
+
+Cell view:
+
+The cell view relies on the matched mapping to provide a template to render.
+
+```HTML
+<div ${ref('wrapper') class="wrapper"}>
+    ${repeat(x => (x.column as TableColumnMapping).mappings,
+        html<Mapping, TableColumnMappingCellView>`
+            ${when((x, c) => x.key === (c.parent as TableColumnMappingCellView).cellRecord.value,
+                html<Mapping>`${x => x.cellViewTemplate}`
+            )}
+        `
+    )}
+</div>
+${when(x => x.wrapper.childElementCount === 0, html<TableColumnMappingCellView>`
+    ${repeat(x => (x.column as TableColumnMapping).mappings,
+        html<Mapping, TableColumnMappingCellView>`
+            ${when((x, c) => x.default,
+                html<Mapping>`${x => x.cellViewTemplate}`
+            )}
+        `
+    )}
+`)}
+```
+
+Group header view:
+
+Similarly, the group header view relies on the matched mapping to provide a template to render.
+
+Note the following requires that `TableColumnMappingGroupHeaderView` has a reference to the column with which it is associated. This is needed to enumerate the column's mapping elements.
```

Comment:
Without looking at the code I'm not sure if this is a good suggestion or not, but did you consider inverting this dependency? Instead of having the header view query the column to find out which mapping to display, make it the column's responsibility to set attributes on each header view so that it has the information it needs to render.

I can think of a couple ways to achieve this:
1. put that information in column internals `columnConfig`, which is generic and is already provided to the header view
2. change `createGroupHeaderViewTemplate()` to be a virtual method on a column instead of a static helper. Then the mapping columns could override it to add additional configuration.

Just some ideas, it might require prototyping to decide if they're feasible and whether we like them better. It just feels wrong for an element to be looking up information in other elements in order to decide what to render.

I would be ok with deferring this discussion until implementation if it's easier to have when referring to code.

Reply:
I've updated the spec accordingly.